core/Loc.cc

```diff
@@ -257,7 +257,11 @@ optional<string_view> Loc::source(const GlobalState &gs) const {
         return nullopt;
     } else {
         auto source = this->file().data(gs).source();
-        return source.substr(beginPos(), endPos() - beginPos());
+        if (endPos() < source.size()) {
+            return source.substr(beginPos(), endPos() - beginPos());
+        } else {
+            return nullopt;
+        }
```

Comment:
1. I think there is an off by one here? Shouldn't `endPos()` be allowed to be equal to `source.size()`?

2. Do we think we can instead make it an invariant that `endPos() <= source.size()`? It seems like we've gotten this far with that being the case, and I'd love to maintain it if possible.

Reply:
1. I believe that's right.
2. I forget why this happened. This fixed a crash. I think it might've been bad argumentName locs created in a rewriter pass?